source/security/authentication.txt

```diff
@@ -0,0 +1,401 @@
+``AssumeRoleWithWebIdentity`` request, ensure that the AWS config file exists in your
+environment and is configured correctly. To learn how to create and configure
+an AWS config file, see `Configuration <https://docs.aws.amazon.com/sdkref/latest/guide/creds-config-files.html>`__
+in the AWS documentation.
+
+After you configure your environment for an ``AssumeRoleWithWebIdentity`` request,
+set the ``authMechanism`` connection option to ``'MONGODB-AWS'``.
+To view an example that sets the ``authMechanism`` option, see the :ref:`authMechanism example
+<php-mongodb-aws-env-example>` on this page.
+
+.. tip::
+
+   For more information about using an ``AssumeRoleWithWebIdentity`` request to
+   authenticate your application, see the following AWS documentation:
+
+   - `Authenticating users for your cluster from an OpenID Connect identity provider
+     <https://docs.aws.amazon.com/eks/latest/userguide/authenticate-oidc-identity-provider.html>`__
+   - `AssumeRoleWithWebIdentity
+     <https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html>`__
+
+.. _php-mongodb-aws-ecs:
+
+ECS Metadata
+~~~~~~~~~~~~
+
+If your application runs in an Elastic Container Service (ECS) container,
+the {+php-library+} can automatically retrieve temporary AWS credentials from an
+ECS endpoint. To do so, specify the URI of the ECS endpoint in an environment variable called
+``AWS_CONTAINER_CREDENTIALS_RELATIVE_URI``, as shown in the following example:
+
+.. code-block:: sh
+
+   export AWS_CONTAINER_CREDENTIALS_RELATIVE_URI=<URI of the ECS endpoint>
+
+After you set the environment variable, set the ``authMechanism``
+connection option to ``'MONGODB-AWS'``. To view an example that sets the
+``authMechanism`` option, see the :ref:`authMechanism example
+<php-mongodb-aws-env-example>` on this page.
+
+.. _php-mongo-aws-ec2:
+
+EC2 Instance Metadata
+~~~~~~~~~~~~~~~~~~~~~
+
+The {+driver-short+} can automatically retrieve temporary AWS credentials from an
+Amazon Elastic Cloud Compute (EC2) instance. To use temporary credentials from
+within an EC2 instance, set the ``authMechanism`` connection option to ``'MONGODB-AWS'``.
+To view an example that sets the ``authMechanism`` option, see the :ref:`authMechanism example
+<php-mongodb-aws-env-example>` on this page.
+
+.. note::
+
+   The {+php-library+} retrieves credentials from an EC2 instance only if the
+   environment variables described in the :ref:`php-mongodb-aws-env-vars` section
+   are not set.
+
+Additional Information
+----------------------
+
+To learn more about creating a ``MongoDB\Client`` object in the {+php-library+},
+see the :ref:`php-client` guide.
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+To learn more about the ``MongoDB\Client`` class, see :phpclass:`MongoDB\Client` 
+in the library API documentation.
+
+To view a full list of URI options that you can pass to a ``MongoDB\Client``, see
+`MongoDB\\Driver\\Manager::__construct Parameters
```

Comment:
```suggestion
`MongoDB\\Driver\\Manager::__construct() Parameters
```

Your call, but I think the parentheses are warranted if we're referring the constructor function.

Reply:
agreed